[winstonlimjy] iP

winstonlimjy opened the pull request:
# Tanjiro Chatbot
> "Walao, this demon is so hard to kill. Nevermind, we have Tanjiro!" - no one ever

Tanjiro Chatbot is about using your keyboard to fight. Today, we will be fighting ourselves!

- **Interactive**: It's super easy to use if you have the commands. ~_PS: for now (maybe)_~ :thinking:
- **Fun**: Instead of talking to yourself, you can talk to me!
- **Engaging**: Tanjiro will always keep you engaged!
- **Safe**: No worries, everything is local. Nothing is saved with me!

**All you need to do is:**
1. Download from [here](https://github.com/nus-cs2103-AY2425S1/ip/pull/326)!
2. Double-click to swing your sword.
3. Add your task and save it.
4. Let Tanjiro settle the rest for you.

**Some basic commands are:**
```
todo ____
deadline ____ /by yyyy-mm-dd hhmm
event ____ /from yyyy-mm-dd hhmm /to yyyy-mm-dd hhmm
find ____
list ____
```

**Features:**
- [x] Managing Tasks
- [ ] Pretty user interface (coming soon) :thinking:
- [ ] Smart (maybe) response (coming soon) :thinking:
- [ ] Reminders (coming soon) :thinking:

If you are a Java programmer, you can use it to practice Java too. Here's the main method:

```ruby
public class Main {
    public static void main(String[] args) {
        Application.launch(MainApp.class, args);
    }
}
```

EVERYTHING IS **FREE OF CHARGE**! ON THE HOUSE!

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

aureliony committed:
build.gradle: Prevent generating a second JAR file

In build.gradle, the dependencies on distZip and/or distTar causes
the shadowJar task to generate a second JAR file for which the
mainClass.set("seedu.duke.Duke") does not take effect.
Hence, this additional JAR file cannot be run.
For this product, there is no need to generate a second JAR file
to begin with.

Let's remove this dependency from the build.gradle to prevent the
shadowJar task from generating the extra JAR file.

winstonlimjy committed:
Level 0. Rename, Greet, Exit

winstonlimjy committed:
Level 2. Add, List

winstonlimjy committed:
Level 3: Mark as Done

winstonlimjy committed:
Level 4: ToDo, Event, Deadline

winstonlimjy committed:
A-TextUiTesting: Automated Text UI Testing

winstonlimjy committed:
Level 5: Handle Errors

winstonlimjy committed:
Level 6: Delete + Fixed Bugs in level 4 + Updated TextUiTesting

winstonlimjy committed:
Fixed extension name of Expected.txt

winstonlimjy committed:
A-Enums

winstonlimjy committed:
Level-7: Save. Changed logic for all actions to accomodate reading/writing from file.

winstonlimjy committed:
Level-8: Dates and Times.

winstonlimjy committed:
A-MoreOOP: Use More OOP. Parser, Storage, Ui, TaskList

winstonlimjy committed:
A-Packages: Organise into Packages.

winstonlimjy committed:
Temporary Text File as DB

winstonlimjy committed:
Merge branch 'add-gradle-support'

winstonlimjy committed:
A-Gradle

winstonlimjy committed:
A-JUnit: Add JUnit Tests for Storage, Parser and DateTimeSystem

winstonlimjy committed:
A-Jar

winstonlimjy committed:
A-JavaDoc: JavaDoc

winstonlimjy committed:
Miscellaneous changes

winstonlimjy committed:
Level-9: Find.

winstonlimjy committed:
Changed Package names

winstonlimjy committed:
A-CodingStandard: Follow the Coding Standard

winstonlimjy committed:
Merge branch 'branch-Level-9'

# Conflicts:
#	src/main/java/Tanjiro.java
#	src/main/java/storage/Storage.java
#	src/main/java/system/DateTimeSystem.java
#	src/main/java/system/Parser.java
#	src/main/java/system/Ui.java

winstonlimjy committed:
Merge branch 'branch-CodingStandard'

# Conflicts:
#	src/main/java/Tanjiro.java
#	src/main/java/commands/TaskList.java
#	src/main/java/storage/Storage.java
#	src/main/java/system/DateTimeSystem.java
#	src/main/java/system/InputProcessor.java
#	src/main/java/system/Parser.java
#	src/main/java/system/Ui.java

winstonlimjy committed:
Changed some OOP.

- Refactored TaskList to Task
- Created a new class TaskList to handle list of tasks
- Reorganise inheritance
- Added Header Comments for all classes except package task

winstonlimjy committed:
Added additional fail cases for date input.

FriedCabbageSalad commented on src/main/java/storage/Storage.java:
```diff
@@ -0,0 +1,167 @@
+        scanner = new Scanner(file);
+        fileWriterAmend = new FileWriter(PATH,true);
+        StringBuilder updated_text = new StringBuilder();
+        while(scanner.hasNextLine()) {
+            String s = scanner.nextLine();
+            if (count == lineNo) {
+                String[] tokens = s.split("\\[_]");
+                String new_line = tokens[0] + "[X]" + tokens[1];
+                updated_text.append(new_line).append("\n");
+            } else {
+                updated_text.append(s).append("\n");
+            }
```
Should a 'while' loop have this format?

FriedCabbageSalad commented on src/main/java/task/Events.java:
```diff
@@ -0,0 +1,52 @@
+    public void addTask(Events e) throws IOException {
+        TaskList.addTasks(e);
+        System.out.println("Got it. I've added this task:");
+        System.out.println("[E][_] " + e.getName() + "(from: " + dateTimeSystem.format(this.start) + " to: " + dateTimeSystem.format(this.end) + ")");
```
Should lines be this long? I noticed this issue elsewhere too.

FriedCabbageSalad commented on src/main/java/task/Task.java:
```diff
@@ -0,0 +1,257 @@
+//                tasks.add(d);
+                TaskList.addTasks(d);
+
+            } else if (s.contains("[E]")){
+                String[] tokens = s.split(" ");
+                StringBuilder name = new StringBuilder();
```
Should if-else class of statements should have the following form?

FriedCabbageSalad reviewed (commented):
LGTM, just some minor nits on coding standard. Keep up the good work!

winstonlimjy committed:
Level 10: Add GUI

- Basic GUI features

winstonlimjy committed:
Added Checkstyle

winstonlimjy committed:
Amended css and designs.

- Extra css
- Bugs not fixed: Ellipsis on list command

winstonlimjy committed:
Fixed GUI bug of text overrun

hogretup commented on src/main/data/test.txt:
```diff
@@ -0,0 +1,6 @@
+1. [D][X] sleep  (by: 2023-12-12 12:12 pm)
+2. [E][_] study  (from: 2023-12-12 12:12 pm to: 2024-12-12 12:12 pm)
+3. [D][X] sleep  (by: 2023-12-12 11:11 am)
+4. [E][X] game  (from: 2023-11-11 11:11 am to: 2024-11-11 12:12 pm)
+5. [T][_] hi
+6. [T][_] omgomgomg gradle
```
should we be pushing data to main branch?

hogretup commented on src/main/java/TanjiroBot/Tanjiro.java:
```diff
@@ -0,0 +1,84 @@
+        storage.createFile();
+        Task.init_list();
+
+        StringBuilder response = new StringBuilder();
+
+        if (parser.containList(userInput)) {
+            response.append(parser.performListTasks());
+        } else if (parser.containMark(userInput)) {
+            response.append(parser.performMark(userInput));
+        } else if (parser.containToDo(userInput)) {
+            response.append(parser.performToDo(userInput));
+        } else if (parser.containDeadline(userInput)) {
+            response.append(parser.performDeadline(userInput));
+        } else if (parser.containEvent(userInput)) {
+            response.append(parser.performEvent(userInput));
+        } else if (parser.containDelete(userInput)) {
+            response.append(parser.performDelete(userInput));
+        } else if (parser.containFind(userInput)) {
+            response.append(parser.performFind(userInput));
+        } else if (parser.containBye(userInput)) {
+            response.append(ui.goodbye());
+        } else {
+            response.append(ui.invalid_input());
+        }
+
+        return response.toString();
```
very neat code!

hogretup commented on src/main/java/system/Ui.java:
```diff
@@ -0,0 +1,205 @@
+            } else {
+                response.append("[D][ ] ").append(t.getName()).append(" (by: ").append(dateTimeSystem.format(t.getDate())).append(")\n");
+            }
+        } else if (t instanceof Events) {
+            if (t.getCurrentStatus() == Task.Status.MARKED) {
+                response.append("[E][X] ").append(t.getName()).append(" (from: ").append(dateTimeSystem.format(t.getStart())).append(" to: ").append(dateTimeSystem.format(t.getEnd())).append(")\n");
+            } else {
+                response.append("[E][ ] ").append(t.getName()).append(" (from: ").append(dateTimeSystem.format(t.getStart())).append(" to: ").append(dateTimeSystem.format(t.getEnd())).append(")\n");
+            }
```
not too sure if possible to perhaps refactor these into a seperate function for redability?

winstonlimjy committed:
Add javadoc to all classes
Enforce coding standard to all classes

winstonlimjy committed:
Add Java assert methods
Add edge-cases handlers

There are several edge cases added to handle the various types of input and dates.
- The comparison of date input with the current date.
- The check of list index number when deleting, marking and unmarking.
- Included assertions in handling dates and input, ensuring edge cases are handled.

winstonlimjy committed:
Add code quality check

Parser.java:
- Add logical structure code
- Edit complicated expressions
- Amend to happy path

Several methods have been amended to ensure that code quality is good. Conditional statements happy path are made more prominent.

winstonlimjy committed:
Merge branch 'branch-A-CodeQuality'

# Conflicts:
#	src/main/data/task_list.txt
#	src/main/java/system/Parser.java

winstonlimjy committed:
Bug Fix: Delete index position

Amended logic of list index position with substring.

winstonlimjy committed:
Add code quality checks

Added addition code quality checks to try out PR on github.

winstonlimjy committed:
Add Stream API

Parser.java: Added additional checks to ensure date format is valid.

Usage of streams api to validate the date format and validity.

winstonlimjy committed:
Merge branch 'master' into branch-A-CodeQuality

winstonlimjy committed:
Merge pull request #1 from winstonlimjy/branch-A-CodeQuality

Add code quality checks

winstonlimjy committed:
Merge branch 'master' into branch-A-Streams

winstonlimjy committed:
Merge pull request #2 from winstonlimjy/branch-A-Streams

Add Stream API

winstonlimjy committed:
Fix Bugs in Parser.java

winstonlimjy committed:
Add B-ViewSchedules extension
Add View Command
Bug Fixes on multiple classes

Parser.java: Fix bugs regarding list, mark, unmark command with empty index.
View command added to view tasks based on a given date and results will be displayed in a list format.
Amended instructions in greetings message as well to give an overview of all commands available.

winstonlimjy committed:
Merge pull request #3 from winstonlimjy/branch-B-ViewSchedules

Add B-ViewSchedules extension

winstonlimjy committed:
Parser.java: Bug Fix on find command, invalid index given.

winstonlimjy committed:
Add custom workflow gradle.yml

winstonlimjy committed:
no message

winstonlimjy committed:
Bug Fix: Storage access from .jar file

winstonlimjy committed:
Removed github workflow to commit first

winstonlimjy committed:
Add gradle.yml

winstonlimjy committed:
Add Gradle.yml

winstonlimjy committed:
Edit gradle.yml

winstonlimjy committed:
Amend code smell

winstonlimjy committed:
Update Better GUI

- Circular Profile Picture
- Asymmetric nature
- Paddings, colors, alignment
- Resizable window

winstonlimjy committed:
Update README.md

winstonlimjy committed:
Add Close Window Function to Bye Command

winstonlimjy committed:
Add Product Screenshot

winstonlimjy committed:
Update README.md
Bug Fix

Bug fixes:
- Ui bugs with from [ ] to [_].
- Remove unnecessary System.out.println

winstonlimjy committed:
Add Unique Personality

New features:
- Hi command
- Displays unique tanjiro response.

winstonlimjy committed:
Add Error Handler
Modify Method Length

winstonlimjy committed:
Update indentation

winstonlimjy committed:
Update JUnit tests

winstonlimjy committed:
Finalize features

Changes made:
- Add Java Docs
- Implement SLAP
- Update error handlers
- Update UI bugs

winstonlimjy committed:
Update README.md

winstonlimjy committed:
Update README.md

winstonlimjy committed:
Create accessor method for Tasklist.java

winstonlimjy committed:
Amend code style

winstonlimjy committed:
Update display message

winstonlimjy committed:
Remove Storage Test

winstonlimjy committed:
Update Parser.java to handle valid time

winstonlimjy committed:
Update README.md

winstonlimjy committed:
Update README.md

winstonlimjy committed:
Update README.md

winstonlimjy committed:
Update Storage Compatibility

winstonlimjy committed:
Update Code Style

winstonlimjy committed:
Updated JavaFX version

winstonlimjy committed:
Update code style

winstonlimjy committed:
Update TaskList.java

winstonlimjy committed:
Update TaskList.java

winstonlimjy committed:
Update GUI

winstonlimjy committed:
Update DialogBoxController
